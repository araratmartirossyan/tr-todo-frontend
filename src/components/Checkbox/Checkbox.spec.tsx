import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { Checkbox } from './index'

interface StateOfTest {
  checked?: boolean
  mockCallback?: jest.Mock
}

describe('Checkbox test', () => {
  const returnState = ({
    checked = false,
    mockCallback = jest.fn(),
  }: StateOfTest) =>
    shallow(<Checkbox checked={checked} onChange={mockCallback} />)

  it('should render Checkbox component', () => {
    const component = returnState({})
    expect(toJson(component)).toMatchSnapshot()
  })

  it('should switch view on click', () => {
    const mockCallback = jest.fn()
    const component = returnState({ mockCallback })
    const input = component.find('[data-test-id="todo-checkbox"]')
    expect(input.exists()).toBe(true)
    input.simulate('change')
    expect(mockCallback.mock.calls.length).toBe(1)
  })
})
