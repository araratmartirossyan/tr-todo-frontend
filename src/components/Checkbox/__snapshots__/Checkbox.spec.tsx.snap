// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox test should render Checkbox component 1`] = `
<styled.div
  checked={false}
  data-test-id="todo-checkbox"
  onChange={[MockFunction]}
>
  <styled.svg
    viewBox="0 0 24 24"
  >
    <polyline
      points="20 6 9 17 4 12"
    />
  </styled.svg>
</styled.div>
`;
